src/main/java/org/folio/service/inventory/InventoryInstanceManager.java

```diff
@@ -381,28 +410,48 @@ public Future<SharingInstance> createShadowInstanceIfNeeded(String instanceId, S
       });
   }
 
-  private Future<String> shareInstanceAmongTenantsIfNeeded(String instanceId, ConsortiumConfiguration consortiumConfiguration,
-                                                           List<Location> locations, RequestContext requestContext) {
-    return findTenantsWithUnsharedInstance(instanceId, locations, requestContext)
-      .map(tenantIds -> tenantIds.stream().map(targetTenantId -> sharingInstanceService.createShadowInstance(instanceId,
-        targetTenantId, consortiumConfiguration, requestContext)).toList())
-      .compose(HelperUtils::collectResultsOnSuccess)
-      .map(sharingInstances -> instanceId);
+  public static String getPublisher(JsonObject instance) {
```

Comment:
if these methods used only in 1 place OrderLinePatchOperationService - it can be declared in that service as private methods. In this case we will avoid static method and simplify contract of InventoryInstanceManager

Reply:
But is it correct that other class imports `INSTANCE_PUBLICATION` field of `InventoryInstanceManager` class and retrieves instance json object field based on it? Idea is that:
1. Instance json representation is part of implementation - other classes should not depend.
2. Method allowing to retrieve field is an abstraction - let's depend on it.
IMHO any changes in `mod-inventory` should lead to updating only `InventoryXyzManager` classes. WDYT?

Reply:
I did not realize before that OrderLinePatchOperationService uses public field from InventoryXyzManager.
As option you can introduce new class InventoryInstanceExtractor and avoid static methods

Reply:
Yes `InventoryUtils` is a better place - we have static methods there.